text/0002-sarif-result-schema.md

```diff
@@ -0,0 +1,1939 @@
+      */
+      shortDescription?: MultiformatMessageString;
+
+      /**
+      * The tool component version, in whatever format the component natively provides.
+      */
+      version?: string;
+
+      /**
+      * Key/value pairs that provide additional information about the tool component.
+      */
+      properties?: PropertyBag;
+  }
+
+  /**
+  * The runtime environment of the analysis tool run.
+  */
+  export interface Invocation {
+      /**
+      * An array of strings, containing in order the command line arguments passed to the tool from the operating
+      * system.
+      *
+      * CHECKUP: args go heree 
+      */
+      arguments?: string[];
+
+      /**
+      * The command line used to invoke the tool.
+      */
+      commandLine?: string;
+
+      /**
+      * The Coordinated Universal Time (UTC) date and time at which the run ended. See "Date/time properties" in the
+      * SARIF spec for the required format.
+      * 
+      * CHECKUP: this property along with startTimeUtc will contain timing info
+      */
+      endTimeUtc?: string;
+
+      /**
+      * The environment variables associated with the analysis tool process, expressed as key/value pairs.
+      * 
+      * CHECKUP: this property will contain the flags, config, configHash 
+      */
+      environmentVariables?: { [key: string]: string };
+
+      /**
+      * Specifies whether the tool's execution completed successfully.
+      */
+      executionSuccessful: boolean;
+
+      /**
+      * The process exit code.
+      */
+      exitCode?: number;
+
+      /**
+      * The machine that hosted the analysis tool run.
+      */
+      machine?: string;
+
+      /**
+      * The Coordinated Universal Time (UTC) date and time at which the run started. See "Date/time properties" in the
+      * SARIF spec for the required format.
+      */
+      startTimeUtc?: string;
+
+      /**
+      * A list of runtime conditions detected by the tool during the analysis.
+      * 
+      * CHECKUP: this is where we will store nonterminal errors collected from Tasks in checkup
+      */
+      toolExecutionNotifications?: Notification[];
+
+
+  }
+
+  export namespace Result {
+      type kind =
+          "notApplicable" |
+          "pass" |
+          "fail" |
+          "review" |
+          "open" |
+          "informational";
+
+      type level =
+          "none" |
+          "note" |
+          "warning" |
+          "error";
+  }
+
+  /**
+  * A result produced by an analysis tool.
+  */
+  export interface Result {
+      /**
+      * An array of 'fix' objects, each of which represents a proposed fix to the problem indicated by the result.
+      * 
+      */
+      fixes?: Fix[];
+
+      /**
+      * A stable, unique identifer for the result in the form of a GUID.
+      */
+      guid?: string;
+      
+      /**
+      * A value that categorizes results by evaluation state.
+      */
+      kind?: Result.kind;
```

Comment:
This field makes me think - I've been looking to add the ability for you to specify whether you want a task to be able to raise and error (errorCode != 0) in order for us to be able to use checkup for validations, for example

- correct lint configuration structure
- correct project configuration structure (think correct Volta setup)

This could be used to populate that?

Reply:
```
    type kind =
        "notApplicable" |
        "pass" |
        "fail" |
        "review" |
        "open" |
        "informational";
```  

This could apply? I think this can be added as a feature in a follow up for sure! Good idea. If you think these types apply for what you are trying to do, lmk and i can create an issue to build that out

Reply:
We're no longer going to add the `errorCode` feature in this way, so we can ignore this!